engine/src/flutter/impeller/geometry/rect.h

```diff
@@ -485,6 +496,31 @@ struct TRect {
     FML_UNREACHABLE();
   }
 
+  /// @brief  Creates a new bounding box that contains this transformed
+  ///          rectangle.
+  ///
+  /// [transform] must be a translate-scale only matrix.
+  [[nodiscard]] constexpr TRect TransformBoundsTranslateScale(
+      const Matrix& transform) const {
+    if (IsEmpty()) {
+      return {};
+    }
+    Scalar tx = transform.m[12];
+    Scalar ty = transform.m[13];
+    Scalar sx = transform.m[0];
+    Scalar sy = transform.m[5];
+    Scalar l = GetLeft() * sx + tx;
+    Scalar r = GetRight() * sx + tx;
+    Scalar t = GetTop() * sy + ty;
+    Scalar b = GetBottom() * sy + ty;
+
+    return TRect<float>::MakeLTRB(std::min(l, r),  //
```

Comment:
Impl idea - another way to do this would be something like:

```
Scalar l,r,t,b;
if (sx > 0) {
  l = left_ * sx + tx;
  r = right_ * sx + tx;
} else {
  l = right_ * sx + tx;
  r = left_ * sx + tx;
}
if (sy < 0) {
...
}
```

not sure if it is any faster.